bluemira/optimisation/_nlopt/optimiser.py

```diff
@@ -146,6 +146,7 @@ def add_eq_constraint(
         f_constraint: OptimiserCallable,
         tolerance: np.ndarray,
         df_constraint: OptimiserCallable | None = None,
+        name: str | None = None,  # noqa: ARG002
```

Comment:
I think it better to make the calls to this explicate in the params passed

(in `_make_optimiser` in _optimise.py)

i.e.

```python
        opt.add_eq_constraint(
            f_constraint=constraint["f_constraint"],
            tolerance=constraint["tolerance"],
            df_constraint=constraint.get("df_constraint", None),
        )
```

Instead of having hanging function params.

Reply:
Done